### PR TITLE
plug the new config in Capsul and LocalEngine objects (#145)

### PR DESCRIPTION
**Pull request by denisri**

### CODE REVIEW

File: capsul/application.py [hunk 1]
```diff
@@ -40,17 +41,13 @@ class Capsul:
...
-            self.config = {}
-        else:
-            if isinstance(config_file, str):
-                self.config_file = Path(config_file)
-            else:
-                self.config_file = config_file
-            with self.config_file.open() as f:
-                self.config = json.load(f)
+    def __init__(self, config_file=None, site_config_file=None):
+        self.config = ApplicationConfiguration('capsul', user_file=config_file,
```
Comment (sapetnioc) [inline capsul/application.py:45]: The name of the application is `capsul` harcoded. It should probably be a parameter of `Capsul` constructor (why not with a `capsul` default value).

File: capsul/engine/local.py [hunk 2]
```diff
@@ -8,13 +8,18 @@
...
+    def __init__(self, label, config=None):
+        self.label = label
         if config is None:
-            self.config = {}
-        else:
+            self.config = EngineConfiguration()
+        elif isinstance(config, EngineConfiguration):
             self.config = config
+        else:
+            self.config = EngineConfiguration()
```
Comment (sapetnioc) [inline capsul/engine/local.py:21]: engines will only be created by internal code. I am not sure we have to support a dictionary value here.

**Review by sapetnioc — APPROVED** [review]

File: capsul/application.py (hunk 1 above)
Comment (denisri) [reply capsul/application.py:45]: Sure, it was set temporarily. I'm changing that

File: capsul/engine/local.py (hunk 2 above)
Comment (denisri) [reply capsul/engine/local.py:21]: I guess so, I have left it to maintain compatibility with the former config dict, but I guess it's useless.

**Pull request merged**